jerry-core/ecma/builtin-objects/ecma-builtin-json.cpp

```diff
@@ -0,0 +1,1840 @@
+
+      /* 2.b.iii */
+      ecma_char_t zt_abbrev[2] = {abbrev, '\0'};
+      ecma_string_t *abbrev_str_p = ecma_new_ecma_string (zt_abbrev);
+
+      tmp_str_p = ecma_concat_ecma_strings (product_str_p, abbrev_str_p);
+      ecma_deref_ecma_string (product_str_p);
+      ecma_deref_ecma_string (abbrev_str_p);
+      product_str_p = tmp_str_p;
+    }
+    /* 2.c */
+    else if (c < 32)
+    {
+      ecma_string_t *backslash_str_p = ecma_get_magic_string (ECMA_MAGIC_STRING_BACKSLASH_CHAR);
+
+      /* 2.c.i */
+      tmp_str_p = ecma_concat_ecma_strings (product_str_p, backslash_str_p);
+      ecma_deref_ecma_string (product_str_p);
+      ecma_deref_ecma_string (backslash_str_p);
+      product_str_p = tmp_str_p;
+
+      /* 2.c.ii */
+      ecma_char_t zt_u_ch[2] = {'u', '\0'};
+      ecma_string_t *u_ch_str_p = ecma_new_ecma_string (zt_u_ch);
+
+      tmp_str_p = ecma_concat_ecma_strings (product_str_p, u_ch_str_p);
+      ecma_deref_ecma_string (product_str_p);
+      ecma_deref_ecma_string (u_ch_str_p);
+      product_str_p = tmp_str_p;
+
+      /* 2.c.iii */
+      ecma_string_t *hex_str_p = ecma_builtin_helper_json_create_hex_digit_ecma_string (c);
+
+      /* 2.c.iv */
+      tmp_str_p = ecma_concat_ecma_strings (product_str_p, hex_str_p);
+      ecma_deref_ecma_string (product_str_p);
+      ecma_deref_ecma_string (hex_str_p);
+      product_str_p = tmp_str_p;
+    }
+    /* 2.d */
+    else
+    {
+      ecma_char_t zt_c[2] = {c, '\0'};
+      ecma_string_t *c_str_p = ecma_new_ecma_string (zt_c);
+
+      tmp_str_p = ecma_concat_ecma_strings (product_str_p, c_str_p);
+      ecma_deref_ecma_string (product_str_p);
+      ecma_deref_ecma_string (c_str_p);
+      product_str_p = tmp_str_p;
+    }
+  }
+
+  MEM_FINALIZE_LOCAL_ARRAY (zt_string_buff);
+
+  /* 3. */
+  tmp_str_p = ecma_concat_ecma_strings (product_str_p, quote_str_p);
+  ecma_deref_ecma_string (product_str_p);
+  ecma_deref_ecma_string (quote_str_p);
+  product_str_p = tmp_str_p;
+
+  /* 4. */
+  return ecma_make_normal_completion_value (ecma_make_string_value (product_str_p));
+} /* ecma_builtin_json_quote */
+
+/**
+ * Abstract operation 'Str' defined in 15.12.3
+ *
+ * See also:
+ *          ECMA-262 v5, 15.12.3
+ *
+ * @return completion value
+ *         Returned value must be freed with ecma_free_completion_value.
+ */
+static ecma_completion_value_t
+ecma_builtin_json_str (ecma_string_t *key_p, /**< property key*/
+                       ecma_object_t *holder_p, /**< the object*/
+                       stringify_context_t *context_p) /**< context*/
+{
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+  ecma_value_t value;
+
+  /* 1. */
+  ecma_completion_value_t get_comp_val = ecma_op_object_get (holder_p, key_p);
+
+  if (!ecma_is_completion_value_normal (get_comp_val))
+  {
+    return get_comp_val;
+  }
+
+  value = ecma_copy_value (ecma_get_completion_value_value (get_comp_val), true);
+  ecma_free_completion_value (get_comp_val);
```

Comment:
In this case I could avoid some (early) returns from an ECMA_TRY_CATCH. But now I use only one return at the end.